calculator.c

```diff
@@ -0,0 +1,148 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+#include <math.h>
+
+#define MAX 100
+
+// Stack structure
+typedef struct {
+    int items[MAX];
+    int top;
+} Stack;
+
+void push(Stack *s, int value) {
+    if (s->top < MAX - 1) {
+        s->items[++(s->top)] = value;
+    } 
+    else {
+        printf("Invalid expression\n");
+        exit(1);
+    }
+}
+
+int pop(Stack *s) {
+    if (s->top >= 0) {
+        return s->items[(s->top)--];
+    } 
+    else {
+        printf("Invalid expression\n");
+        exit(1);
+    }
+}
+
+int peek(Stack *s) {
+    return (s->top >= 0) ? s->items[s->top] : -1;
+}
+
+int precedence(char op) {
+    switch (op) {
+        case '+':
+        case '-': return 1;
+        case '*':
+        case '/': return 2;
+        case '^': return 3;

```

Comment:
the `^` operator was not present in the problem statement. May I know why you have added it?

Reply:
i was just trying out with other operators now i have removed it